index.bs

```diff
@@ -6583,6 +6736,360 @@ This extension enables use of a user verification method.
+
+: Client extension input
+:: <xmp class="idl">
+    dictionary AuthenticationExtensionsDevicePublicKeyInputs {
+        DOMString attestation = "none";
+        sequence<DOMString> attestationFormats = [];
+    };
+
+    partial dictionary AuthenticationExtensionsClientInputs {
+        AuthenticationExtensionsDevicePublicKeyInputs devicePubKey;
+    };
+    </xmp>
+    <div dfn-type="dict-member" dfn-for="AuthenticationExtensionsDevicePublicKeyInputs">
+        :   <dfn>attestation</dfn>
+        ::  The [=[RP]=] MAY use this OPTIONAL member to specify a preference regarding [=attestation conveyance=].
+            Its value SHOULD be a member of {{AttestationConveyancePreference}}.
+            [=Client platforms=] MUST ignore unknown values, treating an unknown value as if the [=map/exist|member does not exist=].
+
+            The default value is {{AttestationConveyancePreference/none}}.
+
+        :   <dfn>attestationFormats</dfn>
+        ::  The [=[RP]=] MAY use this OPTIONAL member to specify a preference regarding the [=attestation=] statement format used by the [=authenticator=].
+            Values SHOULD be taken from the IANA "WebAuthn Attestation Statement Format Identifiers" registry [[!IANA-WebAuthn-Registries]] established by [[!RFC8809]].
+            Values are ordered from most preferable to least preferable.
+            This parameter is advisory and the [=authenticator=] MAY use an attestation statement not enumerated in this parameter.
+
+            The default value is the empty list, which indicates no preference.
+    </div>
+
+    Note: To request the `devicePubKey` extension processing with default options, pass an empty dictionary as the input.
+
+: Client extension processing
+:: If {{AuthenticationExtensionsClientInputs/devicePubKey}} is present, the client creates the authenticator extension input from the client extension input.
+
+: Client extension output
+:: An ArrayBuffer containing the [=device public key attestation object=] that was returned in the authenticator extension output and the signature returned as the [=unsigned extension output=].
+    <xmp class="idl">
+    dictionary AuthenticationExtensionsDevicePublicKeyOutputs {
+        ArrayBuffer authenticatorOutput;
+        ArrayBuffer signature;
+    };
+
+    partial dictionary AuthenticationExtensionsClientOutputs {
+        AuthenticationExtensionsDevicePublicKeyOutputs devicePubKey;
+    };
+    </xmp>
+
+
+: Authenticator extension input
+:: A CBOR expression of the client extension input
+
+    ```
+    devicePublicKeyInputs = {
+      attestation: "none" / "indirect" / "direct" / "enterprise",
+      attestationFormats: [tstr],
+    }
+    $$extensionInput //= (
+      devicePubKey: devicePublicKeyInputs,
+    )
+    ```
+
+: Authenticator extension output
+:: A byte string containing the CBOR encoding of the <dfn>device public key attestation object</dfn>, defined by the `attObjForDevicePublicKey` type:
```

Comment:
As per Issue #1799, I could not read in the way that `clientDataHash` is signed over by a devicePrivateKey.

Reply:
I've added a note highlighting that the RP's challenge is included in the DPK signature in ca1b0c69c2a2e930b25e1d2d7ef2849aef8b9a3a

Reply:
(p.s. just to note that the signature output is in the authenticator's unsigned extension output, rather than the regular extension output, because otherwise the signature would have to sign itself. That's the subsection just below this one.)